app/helpers/job_invocations_helper.rb

```diff
@@ -0,0 +1,115 @@
+      buttons << link_to(_('Rerun'), rerun_job_invocation_path(:id => job_invocation.id),
+                         :class => 'btn btn-default',
+                         :title => _('Rerun the job'))
+    end
+    if authorized_for(hash_for_new_job_invocation_path)
+      buttons << link_to(_('Rerun failed'), rerun_job_invocation_path(:id => job_invocation.id, :failed_only => 1),
+                         :class => 'btn btn-default',
+                         :disabled => !task.sub_tasks.any? { |sub_task| task_failed?(sub_task) },
+                         :title => _('Rerun on failed hosts'))
+    end
+    if authorized_for(:permission => :view_foreman_tasks, :auth_object => task)
+      buttons << link_to(_('Job Task'), foreman_tasks_task_path(task),
+                         :class => 'btn btn-default',
+                         :title => _('See the last task details'))
+    end
+    if authorized_for(:permission => :edit_foreman_tasks, :auth_object => task)
+      buttons << link_to(_('Cancel Job'), cancel_foreman_tasks_task_path(task),
+                         :class => 'btn btn-danger',
+                         :title => _('Try to cancel the job'),
+                         :disabled => !task.cancellable?,
+                         :method => :post)
+    end
+    buttons
+  end
+
+  def job_invocations_buttons
+    [
+      documentation_button_rex('3.2ExecutingaJob'),
+      display_link_if_authorized(_('Run Job'), hash_for_new_job_invocation_path)
+    ]
+  end
+
+  def job_invocation_failed_count(job_invocation)
+    job_invocation.task.sub_tasks.
+      select { |sub_task| task_failed? sub_task }.length
+  end
+
+  def job_invocation_cancelled_count(job_invocation)
+    job_invocation.task.sub_tasks.
+      select { |sub_task| task_cancelled? sub_task }.length
+  end
+
+  def job_invocation_success_count(job_invocation)
+    job_invocation.task.output['success_count'] || 0
+  end
+
+  def job_invocation_pending_count(job_invocation)
+    job_invocation.task.output['pending_count'] ||
+      job_invocation.task.sub_tasks.count
+  end
+
+  def job_invocation_status(invocation)
+    case invocation.status
+      when HostStatus::ExecutionStatus::QUEUED
+        _('queued')
+      when HostStatus::ExecutionStatus::RUNNING
+        _('running %s%') % invocation.progress
+      when HostStatus::ExecutionStatus::OK
+        _('succeeded')
+      when HostStatus::ExecutionStatus::ERROR
+        _('failed')
+      else
+        _('unknown status')
+    end
+  end
+
+  def template_name_and_provider_link(template)
+    template_name = content_tag(:strong, template.name)
+    provider = template.provider.humanized_name
+    link_content = template_name + ' - ' + provider + ' ' +
+      icon_text('edit', '', :kind => 'pficon')
+    link_to_if_authorized(link_content,
+                          hash_for_edit_job_template_path(:id => template.id))
+  end
+
+  def preview_hosts(template_invocation)
+    hosts = template_invocation.targeting.hosts
```

Comment:
this is iterating over all hosts of invocation, it can be thousands of hosts, either add some limit here or revert it to display just one as a temporary solution until we have the selection